### ISSUE TITLE
I got random resultes when query  from a distributed table 

### ISSUE THREAD

**Issue by ALFIE2017** [ClickHouse/ClickHouse#46067]
SELECT *
FROM cluster('cluster_name', system.macros)

Query id: ce9e3255-9d60-4c8d-b123-eaca7d42e108

┌─macro───┬─substitution────┐
│ replica │ cluster01-02-02 │
│ shard   │ 02              │
┌─macro───┬─substitution────┐
│ replica │ cluster01-03-01 │
│ shard   │ 03              │
┌─macro───┬─substitution────┐
│ replica │ cluster01-02-01 │
│ shard   │ 02              │

or 
┌─macro───┬─substitution────┐
│ replica │ cluster01-02-02 │
│ shard   │ 02              │
┌─macro───┬─substitution────┐
│ replica │ cluster01-03-01 │
│ shard   │ 03              │
┌─macro───┬─substitution────┐
│ replica │ cluster01-03-02 │
│ shard   │ 03              │

or 

┌─macro───┬─substitution────┐
│ replica │ cluster01-02-02 │
│ shard   │ 02              │
┌─macro───┬─substitution────┐
│ replica │ cluster01-01-02 │
│ shard   │ 01              │
┌─macro───┬─substitution────┐
│ replica │ cluster01-03-02 │
│ shard   │ 03              │

**Comment by den-crane** [comment]
They should be right.

Please share fragments of .xml with `<macro>` and `<remote_servers>` from ALL 3 SERVERS.

**Comment by ALFIE2017** [comment]
> `<macro>` and `<remote_servers>`

yes, this values are mismatched. I solved this problem by fixed them
